runtime/queries/nix/injections.scm

```diff
@@ -0,0 +1,46 @@
+; mark arbitary languages with a comment
+(
+  ((comment) @injection.language) . [
+    (indented_string_expression (_) @injection.content)
+    (_ (_)* (indented_string_expression (_) @injection.content))
+    (_ (_)* (_ (_)* (indented_string_expression (_) @injection.content)))
+    (_ (_)* (_ (_)* (_ (_)* (indented_string_expression (_) @injection.content))))
+    (_ (_)* (_ (_)* (_ (_)* (_ (_)* (indented_string_expression (_) @injection.content)))))])
+
+((binding
+   attrpath: (attrpath (identifier) @_path)
+   expression: [
+     (indented_string_expression (string_fragment) @injection.content)
+     (_ (_)* (indented_string_expression (string_fragment) @injection.content))
+     (_ (_)* (_ (_)* (indented_string_expression (string_fragment) @injection.content)))
+   ])
+ (#match? @_path "(^\\w*Phase|(pre|post)\\w*|(.*\\.)?\\w*([sS]cript|[hH]ook)|(.*\\.)?startup)$")
+ (#set! injection.language "bash"))
+
+((apply_expression
+   function: (apply_expression function: (_) @_func)
+   argument: (indented_string_expression (string_fragment) @injection.content))
+ (#match? @_func "(^|\\.)writeShellScript(Bin)?$")
+ (#set! injection.language "bash"))
+
+(apply_expression
+  (apply_expression
+    function: (apply_expression
+      function: ((_) @_func
+        (#match? @_func "(^|\\.)runCommand(CC|NoCC|Local|NoCCLocal)?$"))))
+    argument: (indented_string_expression (string_fragment) @injection.content
+      (#set! injection.language "bash")))
+
+(apply_expression
+  function: ((_) @_func
+    (#match? @_func "(^|\\.)writeShellApplication$"))
+  argument: (_ (_)* (_ (_)* (binding
+    attrpath: (attrpath (identifier) @_path
+      (#match? @_path "^text$"))
+    expression: [
+      (indented_string_expression (string_fragment) @injection.content
+        (#set! injection.language "bash"))
+      (_ (_)* (indented_string_expression (string_fragment) @injection.content
+        (#set! injection.language "bash")))
+      (_ (_)* (_ (_)* (indented_string_expression (string_fragment) @injection.content
+        (#set! injection.language "bash"))))]))))
```

Comment:
These two can have their `#match?` and `#set!` predicates moved to the outside of the match too, like the above clauses